packages/medusa/src/types/line-item.ts

```diff
@@ -0,0 +1,17 @@
+import { Cart, Region } from "../models"
+
+export type GenerateInputData = {
+  variantId: string
+  regionId: string
+  quantity: number
+}
+
+export type GenerateContext = {
+  unit_price?: number
+  includes_tax?: boolean
+  metadata?: Record<string, unknown>
+  customer_id?: string
+  order_edit_id?: string
+  region?: Region
+  cart?: Cart
```

Comment:
It is pulling a lot of strings out of the scope of that pr and I don't want to force cast as it can lead to errors if someone else comes after me. So I did a part of that one that I have spread along the way which is specifying the cart props that are expected but those props are still entity base. I think it is a trade-off on that pr to not become too huge.